File: 1D DP/000 Pracitce/Fibonacci/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution{
    public:
        int nth_fib(int n){
            if (n <=1) return n;

            vector<int> dp(n+1);
            dp[0] = 0;
            dp[1] = 1;

            for (int i=2; i<n+1; i++){
                dp[i] = dp[i-1] + dp[i-2];
            }

            return dp[n];
        }
};


int main(){
    Solution solution;
    cout << solution.nth_fib(6); 

    return 0;
}

File: 1D DP/000 Pracitce/Fibonacci/main.py
class Solution:
    def nth_fibonacci(self, n: int):
        if n <= 1:
            return n
        
        dp = [0] * (n+1)
        dp[0] = 0
        dp[1] = 1

        for i in range(2, n+1):
            dp[i] = dp[i-1] + dp[i-2]
        
        return dp[n]

solution = Solution()

n = 7
print(solution.nth_fibonacci(n))